packages/libraries/async-graphql/src/context/mod.rs

```diff
@@ -0,0 +1,32 @@
+use std::time::{Instant, SystemTime, UNIX_EPOCH};
+
+#[cfg(feature = "axum")]
+pub mod axum;
+
+#[derive(Clone, Debug)]
+pub struct GraphQLHiveContext {
+    pub(crate) client_name: Option<String>,
+    pub(crate) client_version: Option<String>,
+    pub(crate) timestamp: u64,
+}
+
+impl GraphQLHiveContext {
+    pub fn new(client_name: Option<String>, client_version: Option<String>) -> Self {
+        Self {
+            client_name,
+            client_version,
+            timestamp: SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
```

Comment:
it's highly unlikely that the system clock is set to a time before the UNIX epoch 😄, but just in case we could handle it a bit more gracefully here.

```suggestion
                    .unwrap_or_else(|err| {
        println!("System time error: {}", err);
        std::time::Duration::from_secs(0)
    })
```